catalog/cataloger_diff.go

```diff
@@ -66,22 +83,23 @@ func (c *cataloger) Diff(ctx context.Context, repository string, leftBranch stri
 	return differences, hasMore, nil
 }
 
-func (c *cataloger) doDiff(ctx context.Context, tx db.Tx, leftID, rightID int64) error {
+func (c *cataloger) doDiff(ctx context.Context, tx db.Tx, leftID, rightID int64, limit int, after string) error {
 	relation, err := getBranchesRelationType(tx, leftID, rightID)
 	if err != nil {
 		return err
 	}
-	return c.doDiffByRelation(ctx, tx, relation, leftID, rightID)
+	return c.doDiffByRelation(ctx, tx, relation, leftID, rightID, limit, after)
```

Comment:
The rule is to limit+1 at the entry function where it supposes to get the additional one record.
So because `Diff` returns "has more" it needs +1.
I'll update the code that will pass newLimit - it should be cleaner.